crates/cargo-test-support/src/lib.rs

```diff
@@ -34,9 +34,18 @@ macro_rules! t {
 
 #[macro_export]
 macro_rules! curr_dir {
-    () => {
-        std::path::Path::new(file!()).parent().unwrap()
-    };
+    () => {{
+        let mut file_path = std::path::Path::new(file!());
+        if !file_path.exists() {
```

Comment:
Would it be possible to move all of this code into a function?  I'm a little concerned about potentially creating thousands of copies of this macro's contents.

Reply:
Done